sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -701,6 +710,21 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
       }
     }
 
+    object ExtractableDateValues {
```

Comment:
why not update `ExtractableValues` to support date？

Reply:
This is because for `InSet` predicate, the `hset`'s value is the int type, we need to use `dateFormatter` to format it, this is the test:
https://github.com/apache/spark/blob/ba2f5531601ff169ff0f9fab9bb60627f65082bf/sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala#L339-L342